HealthcareSystem/Backend/Repository/FeedbackRepository.cs

```diff
@@ -21,18 +21,55 @@ public FeedbackRepository()
             string fileName = "feedback.json";
             path = Path.GetFullPath(fileName);
         }
-        public Model.Users.Feedback NewFeedback(Model.Users.Feedback feedback)
+        public void NewFeedback(Feedback feedback)
         {
             List<Feedback> feedbacks = ReadFromFile();
             feedbacks.Add(feedback);
             WriteInFile(feedbacks);
-            return feedback;
         }
```

Comment:
# Comment

You are working on refactoring backend code, and this is a great moment to apply SOLID. Think about it for a second. What is a problem with methods like this? What will happen when we need to go from FileRepository implementation to NoSQL DB? What would happen if we go to MySQL DB? It is very hard to change every function which has FileRepo to NoSQL repo implementation.

To resolve this, I suggest working on the DIP(Dependency inversion principle). Not just for this FeedbackRepository but for the whole Repository part (for the beginning working on FeedbackRepository is a good start point). Also, look at [REPOSITORY/FACADE](https://www.youtube.com/watch?v=Vb3xx0U21hk) pattern and after you apply this advice, we would have more maintainable and reliability software.

# Concrete example

If we want to change the DataBase from one to another, with this design that is easy. With dependency injection, we can inject in FeedbackServis constructor which Repository implementation we want. And if we want to move from one DataBase to another, we have to inject that another database.

## Design

![image](https://user-images.githubusercontent.com/45834270/97801533-b29abc00-1c3d-11eb-8bce-dccfad0f286c.png)

</br> </br>

## FileRepository implementation for example

![image](https://user-images.githubusercontent.com/45834270/97801671-b5e27780-1c3e-11eb-8887-c93dea105ffe.png)

![image](https://user-images.githubusercontent.com/45834270/97801683-c692ed80-1c3e-11eb-802f-c1a75021a8f2.png)

</br>

## Move to MySQL Repository implementation

![image](https://user-images.githubusercontent.com/45834270/97801714-ec1ff700-1c3e-11eb-8f60-b7a8206b5464.png)

If you see, I only change one line to switch database implementation. Isn't that beautiful.